packages/flame/lib/src/components/component.dart

```diff
@@ -95,8 +291,67 @@ abstract class Component {
   /// ```
   Future<void>? onLoad() => null;
 
+  Future<void>? _onLoadCache;
+
+  /// Since [onLoad] only should run once throughout a components lifetime it is
+  /// cached so that it can be reused when the parent component/game/widget
+  /// changes.
+  Future<void>? get onLoadCache => _onLoadCache ?? (_onLoadCache = onLoad());
```

Comment:
Can't we keep this on the game as it was before? Seems a little bit weird to have a `onLoadCache` getter and if the user overrides this, bad things can happen.

Reply:
If we have it like it was before `onLoad` might be called several times, and then you can't to late final initialization in `onLoad`.

Reply:
We should try to make this an internal thing somehow though. I feel that this method could confuse the user when overriding the `onLoad` method for their own components.

Reply:
If only dart had `protected`... Any ideas for how to make it internal? Since it needs to be reached by the `GameWidget` and from `ComponentSet`.

Reply:
Forgot that `@protected` exists, so I used that now